content/guides/higher-order-functions.adoc

```diff
@@ -0,0 +1,184 @@
+= Higher Order Functions
+Michael Zavarella
+2017-09-15
+:jbake-type: guides
+:toc: macro
+:icons: font
+
+ifdef::env-github,env-browser[:outfilesuffic: .adoc]
+
+toc:[]
+
+== First Class Functions
+
+In functional programming, functions are first class citizens.
+This means functions can be treated as values.
+They can be assigned as values, passed into functions, and returned from functions.
+
+It's common to see function definitions in Clojure using `defn` like `(defn foo ...)`.
+However, this is just syntactic sugar for `(def foo (fn ...))`
+`fn`, and implicitly `defn`, yield a function object as its value that can be passed around like any other Clojure object.
```

Comment:
`defn` does not yield a function object (it returns a var which refers to a function object)